app/src/main/java/com/example/concordia_campus_guide/Activities/PathsViewModel.java

```diff
@@ -57,4 +72,111 @@ public boolean areInSameBuilding(Place from, Place to) {
         }
         return false;
     }
+
+    public void adaptOutdoorDirectionsToInfoCardList(List<DirectionWrapper> directionWrappers) {
+        for (DirectionWrapper dw : directionWrappers) {
+            PathInfoCard card = new PathInfoCard(dw.getDirection().getTransportType(), dw.getDirection().getDuration() / 60, dw.getDirection().getDescription());
+            this.infoCardList.add(card);
+        }
+    }
+
+    public void adaptIndoorDirectionsToInfoCardList(List<WalkingPoint> walkingPointList) {
+        for (int i = 0; i < walkingPointList.size() - 1; i++) {
+            WalkingPoint startWalkingPoint = walkingPointList.get(i);
+            WalkingPoint endWalkingPoint = walkingPointList.get(i + 1);
+            distanceBetweenPoints += getDistanceFromLatLonInKm(startWalkingPoint.getCoordinate().getLatitude(), startWalkingPoint.getCoordinate().getLongitude(), endWalkingPoint.getCoordinate().getLatitude(), startWalkingPoint.getCoordinate().getLongitude());
+            long timeTakenInMinutes;
+            String description = "";
+            PathInfoCard infoCardTypes;
+            if (startWalkingPoint.getPointType() == PointType.CLASSROOM) {
+                description = "Leave classroom";
+                timeTakenInMinutes = (long) (distanceBetweenPoints * 60 / 5);
+                infoCardTypes = new PathInfoCard("Classroom", timeTakenInMinutes, description);
+                infoCardList.add(infoCardTypes);
+                distanceBetweenPoints = 0;
+            } else if (startWalkingPoint.getPointType() == PointType.ENTRANCE) {
+                description = "Walk towards entrance";
+                timeTakenInMinutes = (long) (distanceBetweenPoints * 60 / 5);
+                infoCardTypes = new PathInfoCard("ENTRANCE", timeTakenInMinutes, description);
+                infoCardList.add(infoCardTypes);
+                distanceBetweenPoints = 0;
+            }
+            addIndoorDescriptionToList(startWalkingPoint, endWalkingPoint);
+        }
```

Comment:
both if-else have the same statements as the cases in addIndoorDescriptionToList, if you create a sub-method that does that, it will already be cleaner